src/aap_eda/core/utils/rulebook.py

```diff
@@ -88,3 +89,66 @@ def get_rulebook_hash(rulebook: str) -> str:
     sha256.update(rulebook)
 
     return sha256.hexdigest()
+
+
+def swap_webhook_sources(
+    data: str, webhook_sources: dict, mappings: list[dict]
+) -> str:
+    """Swap out the sources with webhook sources that match the name.
+
+    Preserve the filters if they exist for the source.
+    """
+    rulesets = yaml.safe_load(data)
+    counter = 1
+    current_names = set()
+
+    mapping_dict = {}
+    for mapping in mappings:
+        mapping_dict[mapping["source_name"]] = mapping["webhook_name"]
+
+    for ruleset in rulesets:
+        new_sources = []
+        for source in ruleset.get("sources", []):
+            default_name = f"{DEFAULT_SOURCE_NAME_PREFIX}{counter}"
+            counter += 1
+
+            src_name = source.get("name", default_name)
+            if src_name in current_names:
+                src_name = default_name
+
+            current_names.add(src_name)
+
+            if src_name in mapping_dict:
+                webhook_name = mapping_dict[src_name]
+
+                if webhook_name in webhook_sources:
+                    updated_source = _updated_webhook_source(
+                        webhook_name, source, webhook_sources
+                    )
+                    new_sources.append(updated_source)
+                    LOGGER.debug(
+                        "Source %s updated with Webhook Source", webhook_name
+                    )
+                else:
+                    msg = "No event stream found for source %s", src_name
+                    LOGGER.error(msg)
+                    raise serializers.ValidationError(msg)
```

Comment:
@hsong-rh Is this really needed since we called this after doing the Validation

Reply:
Fixed